cartography/intel/aws/codeguru.py

```diff
@@ -0,0 +1,249 @@
+import json
+import logging
+import pprint
+from typing import Any
+from typing import Dict
+from typing import List
+from urllib import response
+
+import boto3
+import neo4j
+
+from cartography.util import aws_handle_regions
+from cartography.util import run_cleanup_job
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+@aws_handle_regions
+def get_repository_associations(boto3_session: boto3.session.Session, region: str) -> List[str]:
+    client = boto3_session.client('codeguru-reviewer', region_name=region)
+    paginator = client.get_paginator('list_repository_associations')
+    associations: List[Any] = []
+    for page in paginator.paginate():
+        associations.extend(page.get('RepositoryAssociationSummaries', []))
+    return associations
+
+@timeit
+def get_association_names(associations: List[str]) -> List[str]:
+    names: List[Any] = []
+    for assoc in associations:
+        names.append(assoc['Name'])
+
+    return names
+
+def get_recommendation_feedbacks(boto3_session: boto3.session.Session,region: str,code_reviews: List[str]) -> List[str]:
+    client = boto3_session.client('codeguru-reviewer', region_name=region)
+
+    queues: List[Dict] = []
+    for cr in code_reviews:
+        response = client.list_recommendation_feedback(CodeReviewArn=cr['CodeReviewArn'])
+        for queue in response['RecommendationFeedbackSummaries']:
+            queue['Reactions'] = queue['Reactions'][0]
+            queue['RecommendationId'] = queue['RecommendationId']
+            queues.append(queue)
+
+    return queues
+
+@timeit
+@aws_handle_regions
+def get_code_reviews(boto3_session: boto3.session.Session,region: str,association_names: List[str]) -> List[str]:
+    client = boto3_session.client('codeguru-reviewer', region_name=region)
+    response = client.list_code_reviews(States=['Completed'],Type='PullRequest',RepositoryNames=association_names)
+    return response['CodeReviewSummaries']
+
+@timeit
+@aws_handle_regions
+def get_recommendations(boto3_session: boto3.session.Session,region: str,code_reviews: List[str]) -> List[str]:
+    client = boto3_session.client('codeguru-reviewer', region_name=region)
+    recommendations: List[Any] = []
+    queues: List[Dict] = []
+    for cr in code_reviews:
+        response = client.list_recommendations(CodeReviewArn=cr['CodeReviewArn'])
+        for queue in response['RecommendationSummaries']:
+            queue['CodeReviewArn'] = cr['CodeReviewArn']
+            queues.append(queue)
+
+    return queues
+
+@timeit
+def load_repository_associations(
+    neo4j_session: neo4j.Session,
+    data: list,
+    region: str,
+    current_aws_account_id: str,
+    aws_update_tag: int,
+) -> None:
+    cypher = """
+    UNWIND {Repos} AS repo
+        MERGE (cga:CodeguruAssociation{associationarn: repo.AssociationArn})
```

Comment:
`MERGE (cga:CodeguruAssociation{id: repo.AssociationArn})`

By convention, the ID used when creating records is always supposed to be `id`